mypy/newsemanal/semanal.py

```diff
@@ -3874,13 +3871,11 @@ def lookup_qualified(self, name: str, ctx: Context,
                         if (not n
                                 and '__getattr__' in names
                                 and not self.is_incomplete_namespace(namespace)):
+                            fullname = namespace + '.' + '.'.join(parts[i:])
                             gvar = self.create_getattr_var(names['__getattr__'],
-                                                           parts[i], parts[i])
+                                                           parts[i], fullname)
                             if gvar:
                                 n = SymbolTableNode(GDEF, gvar)
-                                # TODO: Enable this (the purpose is likely to preserve the
-                                #       identity of the Var node if there are multiple imports)
-                                # names[name] = n
```

Comment:
It looks like my suggestion about inserting the `Var` is about this. If it doesn't work for some reason, this should be explained in a comment.